src/plugins/visualizations/public/visualize_app/app.tsx

```diff
@@ -45,6 +86,46 @@ export const VisualizeApp = ({ onAppLeave }: VisualizeAppProps) => {
     // so the global state is always preserved
   }, [query, kbnUrlStateStorage, pathname]);
 
+  useEffect(() => {
+    const checkESOrDataViewExist = async () => {
+      // check if there is any data view or data source
+      const hasUserDataView = await dataViews.hasData.hasUserDataView().catch(() => false);
+      const hasEsData = await dataViews.hasData.hasESData().catch(() => true);
```

Comment:
I think this should also be `false`. If there was an exception, we probably want to default to `false` rather than to `true`. I assume this was taken from my Discover PR but I had also changed it in the meantime: 
https://github.com/elastic/kibana/blob/main/src/plugins/discover/public/application/main/discover_main_route.tsx#L73

Reply:
Done

Reply:
I wonder then if the `hasESData()` method should not be the one providing the `catch` and default value for all consumers.